bundles/org.openhab.binding.myq/src/main/java/org/openhab/binding/myq/internal/handler/MyQAccountHandler.java

```diff
@@ -0,0 +1,330 @@
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.myq.internal.handler;
+
+import static org.openhab.binding.myq.internal.MyQBindingConstants.*;
+
+import java.util.Random;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.eclipse.jetty.client.HttpClient;
+import org.eclipse.jetty.client.api.ContentProvider;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.api.Result;
+import org.eclipse.jetty.client.util.BufferingResponseListener;
+import org.eclipse.jetty.client.util.StringContentProvider;
+import org.eclipse.jetty.http.HttpMethod;
+import org.eclipse.jetty.http.HttpStatus;
+import org.openhab.binding.myq.internal.MyQConfiguration;
+import org.openhab.binding.myq.internal.MyQDiscoveryService;
+import org.openhab.binding.myq.internal.dto.AccountDTO;
+import org.openhab.binding.myq.internal.dto.ActionDTO;
+import org.openhab.binding.myq.internal.dto.DevicesDTO;
+import org.openhab.binding.myq.internal.dto.LoginRequestDTO;
+import org.openhab.binding.myq.internal.dto.LoginResponseDTO;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.ThingTypeUID;
+import org.openhab.core.thing.ThingUID;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.ThingHandler;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.gson.FieldNamingPolicy;
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+
+/**
+ * The {@link MyQAccountHandler} is responsible for communicating with the MyQ API based on an account.
+ *
+ * @author Dan Cunningham - Initial contribution
+ */
+@NonNullByDefault
+public class MyQAccountHandler extends BaseBridgeHandler {
+    private static final String BASE_URL = "https://api.myqdevice.com/api";
+    private final Logger logger = LoggerFactory.getLogger(MyQAccountHandler.class);
+    private final Gson gsonUpperCase = new GsonBuilder().setFieldNamingPolicy(FieldNamingPolicy.UPPER_CAMEL_CASE)
+            .create();
+    private final Gson gsonLowerCase = new GsonBuilder()
+            .setFieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES).create();
+    private @Nullable Future<?> normalPollFuture;
+    private @Nullable Future<?> rapidPollFuture;
+    private @Nullable String securityToken;
+    private @Nullable AccountDTO account;
+    private MyQDiscoveryService discoveryService;
+    private Integer normalRefreshInterval = 60;
+    private Integer rapidRefreshInterval = 5;
```

Comment:
Primitive types should be used where possible.

It's good practice to append the unit to the field name e.g. TIMEOUT_SEC.

Can these be static final?
```suggestion
    private int normalRefreshInterval = 60;
    private int rapidRefreshInterval = 5;
```